src/libraries/System.Reflection.MetadataLoadContext/tests/src/Tests/Type/TypeTests.cs

```diff
@@ -464,6 +464,37 @@ public static void TestIsValueType()
             return;
         }
 
+        public static IEnumerable<object[]> TypeDescTypesIsPublic_TestData()
+        {
+            yield return new object[] { typeof(int).Project().MakeByRefType(), true, true };
+            yield return new object[] { typeof(int).Project().MakePointerType(), true, true };
+            yield return new object[] { typeof(int).Project(), true, true };
+            yield return new object[] { typeof(PubilcClass.InternalNestedClass).Project().MakeByRefType(), true, false };
+            yield return new object[] { typeof(PubilcClass.InternalNestedClass).Project().MakePointerType(), true, false };
+            yield return new object[] { typeof(PubilcClass.InternalNestedClass).Project(), false, false };
+            yield return new object[] { typeof(PubilcClass.InternalNestedGenericClass<>).Project().MakeGenericType(typeof(PubilcClass).Project()), false, false };
+            yield return new object[] { typeof(PubilcClass).Project().MakeByRefType(), true, true };
+            yield return new object[] { typeof(PubilcClass).Project().MakePointerType(), true, true };
+            yield return new object[] { typeof(PubilcClass).Project(), true, true };
+        }
+
+        [Theory]
+        [MemberData(nameof(TypeDescTypesIsPublic_TestData))]
+        public static void TypeDescTypesIsPublic(Type type, bool isPublic, bool isVisible)
+        {
+            Assert.Equal(isPublic, type.IsPublic);
+            Assert.Equal(!isPublic, type.IsNestedAssembly);
+            Assert.Equal(isVisible, type.IsVisible);
+        }
+
+        [Fact]
+        [SkipOnMono("System.TypeLoadException : Could not find type 'System.MonoFNPtrFakeClass'")]
```

Comment:
What is the `MonoFNPtrFakeClass` type? Is this a known bug?

Reply:
> What is the MonoFNPtrFakeClass type?

Not sure, only one hit found, seems function pointer name created in mono, 
https://github.com/dotnet/runtime/blob/053a472be0143bda0891c715d7b2ce7398538127/src/mono/mono/metadata/class-init.c#L1541

> Is this a known bug?

Guess not know, but seems [function pointer is not supported in MLC](https://github.com/dotnet/runtime/issues/43791), probably i should not have added this test in MLC

Reply:
> Not sure, only one hit found, seems function pointer name created in mono,

There should be no way for MLC to hit Mono runtime reflection implementation. Either this condition is bogus; or there is something really wrong.

Reply:
Here is [full log](https://helix.dot.net/api/2019-06-17/jobs/56c812f4-26b8-46ae-b108-e837c360f2fe/workitems/System.Reflection.MetadataLoadContext.Tests/console), it hit mono runtime
```log
System.Reflection.Tests.TypeTests.FunctionPointerTypeIsPublic [FAIL]
      System.TypeLoadException : Could not find type 'System.MonoFNPtrFakeClass' in assembly '/private/tmp/helix/working/A5E8094B/p/shared/Microsoft.NETCore.App/7.0.0/System.Private.CoreLib.dll'.
      Stack Trace:
        /_/src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/Helpers.cs(298,0): at System.Reflection.TypeLoading.Helpers.LoadTypeFromAssemblyQualifiedName(String name, RoAssembly defaultAssembly, Boolean ignoreCase, Boolean throwOnError)
        /_/src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Assemblies/RoAssembly.cs(106,0): at System.Reflection.TypeLoading.RoAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase)
        /_/src/libraries/System.Reflection.MetadataLoadContext/tests/src/TestUtils/TestUtils.JittedRuntimes.cs(40,0): at System.Reflection.Tests.TestUtils.<>c__DisplayClass20_0.<Project>b__1(Type t)
        /_/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs(1146,0): at System.Collections.Concurrent.ConcurrentDictionary`2[[System.Type, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Type, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].GetOrAdd(Type key, Func`2 valueFactory)
        /_/src/libraries/System.Reflection.MetadataLoadContext/tests/src/TestUtils/TestUtils.JittedRuntimes.cs(40,0): at System.Reflection.Tests.TestUtils.Project(Type type)
        /_/src/libraries/System.Reflection.MetadataLoadContext/tests/src/Tests/Type/TypeTests.cs(493,0): at System.Reflection.Tests.TypeTests.FunctionPointerTypeIsPublic()
        /_/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.Mono.cs(386,0): at System.Reflection.RuntimeMethodInfo.InvokeWorker(Object obj, BindingFlags invokeAttr, Span`1 parameters)
  Finished:    System.Reflection.MetadataLoadContext.Tests
```

Reply:
Ok, so this is actually a problem in the test helper method. The representation of function pointers is broken in reflection stack (https://github.com/dotnet/runtime/issues/11354), but it is broken in a different way on CoreCLR vs. Mono.

I think the test should be disabled against https://github.com/dotnet/runtime/issues/11354. It may be a good idea to disable it unconditionally. The test happens to pass outside Mono currently, but it is not testing what its name suggests there.